spec/models/rook_spec.rb

```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe Rook, type: :model do
+  it 'has a valid Factory' do
+    build(:rook).expect be_valid
```

Comment:
You're right - total rabbit hole.

Good news though, you're validating this factory right.

Bad news is we didn't need any of the pieces or anything - although I recommend you take the Pieces STI ticket, because you can use a lot of this code there.

We only need factories for the schema that currently exists; no migrations should be required. https://github.com/teamdeepblue/caissa-chess/blob/master/db/schema.rb